fastify.js

```diff
@@ -282,11 +282,13 @@ function build (options) {
     this.context = context
   }
 
-  function hookIterator (fn, state, next) {
+  function hookIterator (fn, state, next, release) {
+    if (state.res.finished === true) return release()
     return fn(state.req, state.res, next)
   }
 
   function middlewareCallback (err, state) {
+    if (state.res.finished === true) return
```

Comment:
Same thing in `onRunMiddlewares`

Reply:
There should be no need of that test. The middlewares are handled by middie, and the `res.finished` check is performed before to call the iterator or `complete`. :)
https://github.com/fastify/middie/blob/master/middie.js#L70-L77

Reply:
Oh that's included in `middie`. Cool!